dist/systemd/system/zincati.service

```diff
@@ -8,6 +8,10 @@ After=network.target
 # this prevents rolling out broken updates to all nodes in the fleet.
 Requires=boot-complete.target
 After=boot-complete.target
+# Make sure we don't inadvertently reboot the system before a machine-id is
+# created so that we don't cause ConditionFirstBoot=true units to run twice
+# See discussions in https://github.com/systemd/systemd/issues/4511.
+After=systemd-machine-id-commit.service
```

Comment:
Shouldn't we use `After=first-boot-complete.target` here?

Reply:
That's not strong enough because you might end up running between `first-boot-complete.target` and `systemd-machine-id-commit.service`. The order should basically be:

1. stuff which use `ConditionFirstBoot=true`
2. `first-boot-complete.target`
3. `systemd-machine-id-commit.service`
4. stuff which wants to reboot the node

Reply:
For completeness, I verified that `systemd-machine-id-commit.service` is a oneshot and that it has a `After=first-boot-complete.target` dependency (in the upcoming systemd release): https://github.com/systemd/systemd/blob/aa5502bb336fdc4546e564c60d36b3d4e7909bc7/units/systemd-machine-id-commit.service